test/templates/network-test.yaml.tpl

```diff
@@ -22,10 +22,18 @@ spec:
         - /bin/bash
         - -c
         - |
-          apt-get update && apt-get install -y netcat-openbsd iptables python3 python3-pip; pip3 install --break-system-packages scapy; while true; do sleep 30; done
+          apt-get update && apt-get install -y netcat-openbsd iptables iproute2 ipset python3 python3-pip; pip3 install --break-system-packages scapy; while true; do sleep 30; done
         securityContext:
           privileged: true
 
+        readinessProbe:
+          exec:
+            command:
+            - /bin/bash
+            - -c
+            - |
+              which nc && which iptables-legacy && which ip && which ipset && which python3
```

Comment:
I'm wondering if this won't break someday. 
We install a packet iptables in a container with the latest ubuntu image and call the iptables-legacy executable.
I could imagine, that at one point in time we won't get iptables-legacy.
It was like this already before.
Should we maybe fix it, before we run into failing tests?

Reply:
The alternative is to build our own image for this. Or move to nftables which is a bigger change but should be done anyways. We could pin the ubuntu version in the meantime? I'd rather move this into its own story tbh.

Reply:
Why is the `readiness` probe necessary? How does this avoid random script failures?

The `pip3` installation of `scapy` is not checked. Should it be checked?

Reply:
@ScheererJ the [code](https://github.com/domdom82/gardener-extension-shoot-networking-filter/blob/blackhole-wg/test/e2e/network_test.go#L110) in the test suite waits until the daemonset is ready. Without the readiness probe this will be true as soon as the pod it running. The problem is that by that time we could still be executing the `apt-get` commands to install the tools needed in the [network-filter-test.sh](https://github.com/domdom82/gardener-extension-shoot-networking-filter/blob/blackhole-wg/test/e2e/network_test.go#L126) script.

It all depends on the speed of how fast the container is set up - which it should not. That's why there were several sleeps all over the code, to wait until (probably) the setup was done. The proper way is to have the daemonset only report ready if it actually is ready to do its job (= call the `network-filter-test.sh` script) which is far more reproducible than arbitrary sleeps in the code.

Yes, we could add a test for the installation of scapy for even more safety.

Reply:
Added check for scapy, too.